core/src/main/scala/kafka/zk/KafkaZkClient.scala

```
@@ -619,6 +664,224 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends
+    }
+  }
+
+  /**
+   * Deletes the Acl change notifications associated with the given sequence numbers.
+   * @param sequenceNumbers the sequence numbers associated with the Acl change notifications to be deleted.
+   */
+  def deleteAclChangeNotifications(sequenceNumbers: Seq[String]): Unit = {
+    val deleteRequests = sequenceNumbers.map { sequenceNumber =>
+      DeleteRequest(AclChangeNotificationSequenceZNode.path(sequenceNumber), ZkVersion.NoVersion)
+    }
+    retryRequestsUntilConnected(deleteRequests)
+  }
+
+  /**
+   * Deletes Acl change notification associated with the given sequence number
+   * @param sequenceNumber
+   * @return delete status
+   */
+  def deleteAclChangeNotification(sequenceNumber: String): Boolean = {
+    val deleteRequest = DeleteRequest(AclChangeNotificationSequenceZNode.path(sequenceNumber), ZkVersion.NoVersion)
+    val deleteResponse = retryRequestUntilConnected(deleteRequest)
+    if (deleteResponse.resultCode == Code.OK || deleteResponse.resultCode == Code.NONODE) {
+      true
+    }else {
+      throw deleteResponse.resultException.get
+    }
+  }
+
+  /**
+   * Gets the resource types
+   * @return list of resource type names
+   */
+  def getResourceTypes(): Seq[String] = {
+    getChildren(AclZNode.path)
+  }
+
+  /**
+   * Gets the resource names for a give resource type
+   * @param resourceType
+   * @return list of resource names
+   */
+  def getResourceNames(resourceType: String): Seq[String] = {
+    getChildren(ResourceTypeZNode.path(resourceType))
+  }
+
+  /**
+   * Deletes the given Resource node
+   * @param resource
+   * @return delete status
+   */
+  def deleteResource(resource: Resource): Boolean = {
+    deleteRecursive(ResourceZNode.path(resource))
+    !resourceExists(resource)
+  }
+
+  /**
+   * checks the resource existence
+   * @param resource
+   * @return existence status
+   */
+  def resourceExists(resource: Resource): Boolean = {
+    pathExists(ResourceZNode.path(resource))
```

This is not introduced in this patch, but pathExists() should throw an exception if the result code is not ok. Could you fix it in this patch too?